src/suites/cts/copyImageBitmapToTexture.spec.ts

```diff
@@ -0,0 +1,160 @@
+            break;
+          }
+          failedPixels++;
+          lines.push(`at [${indexExp}], expected ${exp[indexExp]}, got ${actual[indexActual]}`);
+        }
+      }
+      if (failedPixels > 4) {
+        lines.push('... and more');
+        break;
+      }
+    }
+    return this.handleErrorMessage(actual, exp, size, failedPixels, lines);
+  }
+
+  // Using drawImage to extract imageBitmap content.
+  imageBitmapToData(imageBitmap: ImageBitmap): ArrayBufferView {
+    const imageCanvas = document.createElement('canvas');
+    imageCanvas.width = imageBitmap.width;
+    imageCanvas.height = imageBitmap.height;
+
+    const imageCanvasContext = imageCanvas.getContext('2d');
+    if (!imageCanvasContext) {
+      throw new Error('Cannot create canvas context for reading back contents from imageBitmap.');
+    }
+
+    imageCanvasContext.drawImage(imageBitmap, 0, 0, imageBitmap.width, imageBitmap.height);
+    return imageCanvasContext.getImageData(0, 0, imageBitmap.width, imageBitmap.height).data;
+  }
+}
+
+export const g = new TestGroup(F);
+
+g.test('from image element', async t => {
+  const { width, height, bytesPerPixel } = t.params;
+  const imagePixels = new Uint8ClampedArray(bytesPerPixel * width * height);
+  for (let i = 0; i < width * height * bytesPerPixel; ++i) {
+    imagePixels[i] = i % 4 === 3 ? 255 : i % 256;
+  }
+
+  const imageData = new ImageData(imagePixels, width, height);
+  const imageBitmap = await createImageBitmap(imageData);
+
+  const dst = t.device.createTexture({
+    size: {
+      width: imageBitmap.width,
+      height: imageBitmap.height,
+      depth: 1,
+    },
+    format: 'rgba8uint',
+    usage: GPUTextureUsage.COPY_DST | GPUTextureUsage.COPY_SRC,
+  });
+
+  t.device.defaultQueue.copyImageBitmapToTexture(
+    {
+      imageBitmap,
+      origin: {
+        x: 0,
+        y: 0,
+      },
+    },
+    {
+      texture: dst,
+    },
+    {
+      width: imageBitmap.width,
+      height: imageBitmap.height,
+      depth: 1,
+    }
+  );
+
+  const data = t.imageBitmapToData(imageBitmap);
+
+  // The texture format is r8uint, so the bytes per pixel is 4.
+  const rowPitchValue = calculateRowPitch(imageBitmap.width, bytesPerPixel);
+  const testBuffer = t.device.createBuffer({
+    size: rowPitchValue * imageBitmap.height,
+    usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST,
+  });
+
+  const encoder = t.device.createCommandEncoder();
+
+  encoder.copyTextureToBuffer(
+    { texture: dst, mipLevel: 0, origin: { x: 0, y: 0, z: 0 } },
+    { buffer: testBuffer, rowPitch: rowPitchValue, imageHeight: 0 },
+    { width: imageBitmap.width, height: imageBitmap.height, depth: 1 }
+  );
+  t.device.defaultQueue.submit([encoder.finish()]);
+
+  t.checkCopyImageBitmapResult(
+    testBuffer,
+    data,
+    imageBitmap.width,
+    imageBitmap.height,
+    bytesPerPixel
+  );
+}).params(
+  pcombine(poptions('width', [4, 15]), poptions('height', [1, 2]), poptions('bytesPerPixel', [4]))
```

Comment:
nit: bytesPerPixel can't actually take any values other than 4 right now, so I'm just going to remove it from the params.